flink-runtime/src/main/java/org/apache/flink/runtime/state/SharedStateRegistry.java

```diff
@@ -269,65 +143,23 @@ public String toString() {
         }
     }
 
-    /** The result of an attempt to (un)/reference state */
-    public static class Result {
-
-        /** The (un)registered state handle from the request */
-        private final StreamStateHandle reference;
+    SharedStateRegistry NO_OP =
+            new SharedStateRegistry() {
+                @Override
+                public void close() {}
```

Comment:
Can you move adding the `NO_OP` implementation to the commit that starts using it?

Reply:
Sure, moved it to `[FLINK-24611] Register shared state on checkpoint ACK`.